File: 재귀/SWEA_1217_거듭제곱/solution.cpp
#include <iostream>
using namespace std;
int tc;
int N;
int M;
 
int solve(int n, int m, int now) {
    if (m==0) {
        return now;
    }
    return solve(n,m-1,now*n);
}
int main() {
    int T = 10;
    for (int i=0; i<T; i++) {
        cin >> tc;
        cin >> N >> M;
        int ans = solve(N,M,1);
        cout << "#" << tc << " " << ans << "\n";
    }
    return 0;
}

File: 재귀/SWEA_1217_거듭제곱/Solution.java
package 재귀.SWEA_1217_거듭제곱;

import java.util.Scanner;
 
class Solution
{
    public static Integer solve(int N, int M, int now) {
        if (M == 0) {
            return now;
        }
        return solve(N, M-1, now*N);
    }
    public static void main(String args[]) throws Exception
    {   
        Scanner sc = new Scanner(System.in);
        int T = 10;
        for (int i=0; i<T; i++) {
            int tc = sc.nextInt();
            int N = sc.nextInt();
            int M = sc.nextInt();
 
            int ans = solve(N,M,1);
            System.out.println("#"+tc + " " + ans);
        }
        sc.close();
    }
}